.travis.yml

```diff
@@ -37,6 +37,16 @@ env:
     # @link https://github.com/WordPress/WordPress
     - WP_VERSION=5.0 WP_MULTISITE=0
 
+before_install:
+  # Install old Node since old NPM won't work properly on new Node.
+  - nvm install 4
+  # Update NPM.
+  - npm install -g npm
+  # Reinstall new Node.
+  - nvm install 10
```

Comment:
The above roundabout downgrading/upgrading is actually required. The Travis container by default comes with Node 10 and NPM 1.4.28.

Trying to install Stylelint with NPM this old fails:
https://travis-ci.org/atanas-angelov-dev/twentynineteen/jobs/448533664

Trying to update this old version of NPM on the newer Node 10 causes NPM to break:
https://travis-ci.org/atanas-angelov-dev/twentynineteen/jobs/448538317

So the solution is:
1. Downgrade Node so a version that NPM 1.x works with.
2. Upgrade NPM to the latest version.
3. Upgrade Node back to 10.x. (I switched this to `--lts` instead of 10 specifically)

One improvement I made now was to only do this if NPM's version is exactly 1.4.28 so if the NPM version changes in the container (due to travis changes or distro changes) it will not cause trouble (no new 1.x version is supposed to be shipped so we don't have to do fancy version checking).

Reply:
> The above roundabout downgrading/upgrading is actually required. The Travis container by default comes with Node 10 and NPM 1.4.28.

The Travis CI container comes with Node.js `0.10.36` (Note the leading `0.`)
- https://travis-ci.org/atanas-angelov-dev/twentynineteen/jobs/448533664#L47

> Trying to install Stylelint with NPM this old fails:

Which is why you install Node.js 10.x before updating the npm install:
```yaml
  # Install latest LTS Node.
  - nvm install 10
  # Update NPM.
  - npm install -g npm
```

> Trying to update this old version of NPM on the newer Node 10 causes NPM to break:

Here you are trying to install the latest npm on Node.js `0.10.36` and not on the latest `10.13.0`

> One improvement I made now was to only do this if NPM's version is exactly 1.4.28 so if the NPM version changes in the container (due to travis changes or distro changes) it will not cause trouble (no new 1.x version is supposed to be shipped so we don't have to do fancy version checking).

Can you remove this change please, we will always use the latest release of npm, this is regardless of the Node.js version used so this change is not required.

Reply:
**0**.10 :man_facepalming:

Reply:
I've updated the PR - latest code does this:
- Update NVM so it supports the `--lts` option
- Install Node with --lts
- Update NPM  - this may not be necessary as the LTS install already brings in a modern NPM version.